core/transaction/validation.go

```diff
@@ -0,0 +1,19 @@
+package transaction
+
+import "context"
+
+type Tx interface {
+	Hash() [32]byte // TODO evaluate if 32 bytes is the right size & benchmark overhead of hashing instead of using identifier
```

Comment:
I think we could do:

```go
type Tx[I comparable] interface {
    Identifier() I
}
```


This basically means that the "hash" is an identifier, this identifier can be anything which in go is comparable (you can do ==, <, > operation on it). 

The bad thing is that it would leak one extra generic down the road.

Reply:
What's the problem of just forcing a byte array and using hash as the identifier. Surely, all transactions have a hash.